### PR TITLE
Introduce encoding for query_reply and publisher

### PR DESCRIPTION
**Pull request by Mallets**

### CODE REVIEW

File: include/zenoh-pico/api/primitives.h [hunk 1]
```diff
@@ -317,6 +317,14 @@ const char *zp_config_get(z_config_t *config, unsigned int key);
...
 
+/**
+ * Constructs a known :c:type:`z_encoding_t`.
+ *
+ * Returns:
+ *   Returns the constructed :c:type:`z_encoding_t`.
+ */
+z_encoding_t z_encoding(z_encoding_prefix_t prefix);
+
 /**
```
Comment (cguimaraes) [inline include/zenoh-pico/api/primitives.h:328]: If we are providing the helpers for the encodings, then I would provide one to set the suffix as well.
To be considered if we also need the getters for the `prefix` and `suffix` in  `z_encoding_t`.

And in this case, both `prefix` and `suffix` in  `z_encoding_t` can be privatized.

File: src/api/api.c [hunk 2]
```diff
@@ -122,10 +122,12 @@ int8_t zp_config_insert(z_config_t *config, unsigned int key, z_string_t value)
     return _zp_config_insert(config, key, value);
 }
 
-z_encoding_t z_encoding_default(void) {
-    return (_z_encoding_t){.prefix = Z_ENCODING_PREFIX_EMPTY, .suffix = _z_bytes_make(0)};
+z_encoding_t z_encoding(z_encoding_prefix_t prefix) {
+    return (_z_encoding_t){.prefix = prefix, .suffix = _z_bytes_make(0)};
 }
```
Comment (cguimaraes) [inline src/api/api.c:127]: To be consistent with other data structures at the API-level, I would still keep the `z_encoding_t z_encoding_default(void)`.

**Review by cguimaraes — CHANGES_REQUESTED** [review]
Small changes suggested.

File: src/api/api.c (hunk 2 above)
Comment (p-avital) [reply src/api/api.c:127]: As long as `z_encoding_default` keeps existing (it does, l129, I almost missed it), I'm good with this

Comment (Mallets) [reply src/api/api.c:127]: Indeed, `z_econding_default` has not been remove: just reworked.

File: include/zenoh-pico/api/primitives.h (hunk 1 above)
Comment (Mallets) [reply include/zenoh-pico/api/primitives.h:332]: I've extended `z_encoding` to accept both `prefix` and `suffix` in the constructor. I believe there is no need to add getters for `z_encoding_t`.

**Comment by cguimaraes** [comment]
Regarding the last commit, completeness (considering it is the N parameter on the complete) should be an integer and not a bool. Unless, it has been discussed to be changed while I was off.

**Comment by Mallets** [comment]
It is just misname, `complete` is a bool. `completeness` is still a `z_int_t`.

**Review by cguimaraes — APPROVED** [review]

**Pull request merged**